src/dashboard/Data/Browser/Browser.react.js

```diff
@@ -320,9 +321,21 @@ export default class Browser extends DashboardView {
     return data;
   }
 
+  async fetchParseDataCount(source, filters) {
+    const query = queryFromFilters(source, filters);
+    const count = await query.count({ useMasterKey: true });
```

Comment:
You raise a great point.  

Besides relying on the parse-server timeout, we could also add a restriction on when the count will update and only update if the total number of objects in the table is less than a certain constant.  Not super excited about this option, though.  Might be best to rely on the server's timeout.